### DISCUSSION TITLE
How do you change the start config file for Nginx through Homebrew?

### DISCUSSION THREAD

**Discussion by DanielCoder834** [Homebrew/discussions#5839] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.14
ORIGIN: https://github.com/Homebrew/brew
HEAD: f84082963da8af8c9ccd6dffbe932eba457e2b78
Last commit: 30 hours ago
Branch: stable
Core tap JSON: 31 Dec 18:22 UTC
Core cask tap JSON: 31 Dec 18:22 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.47.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!
```

### Description of issue

I have been trying to run: `brew services start nginx --file forward.conf`, but everytime I do, I get this error:  
```Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/danielkaplan/Library/LaunchAgents/homebrew.mxcl.nginx.plist` exited with 5.
```

According to the `brew services list nginx`, this line says you can change the config file: 
`--file  Use the service file from this location to start the service.`
I am not sure what the issue is and how to resolve it. 
I found this [issue](https://github.com/Homebrew/homebrew-core/issues/98253) to be similar, but I don't have a `/usr/local/var/log/nginx/` folder. 
What is the command for selecting a different start-up config file for Nginx through Homebrew, and what does the error above mean?

**Comment by SMillerDev** [comment]
Why do you have a service plist file with `.conf` as an extension?

**Reply by DanielCoder834** [reply]
I don't to want to touch the default as I can use it as an example when I get stuck. Are you not supposed to use a .conf file for Nginx configuration? 

I didn't create or modify that plist file. I think it was downloaded when I brew installed Nginx.

edit: What is the difference between the plist file and conf files?

**Reply by DanielCoder834** [reply]
This is the contents of the plist file: 
```
http {
    server {
        listen 8080;

        location / {
            proxy_pass http://$http_host$request_uri;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
    }
}
```

**Reply by alebcay** [reply, marked as answer]
Perhaps the documentation/wording can be clarified here, but the `--file` argument expects a configuration file in plist format for macOS launchctl, _not_ an application-specific configuration file (e.g. nginx conf file).

If you're more familiar with init systems on Linux, it's like the argument allows you to provide a custom systemd unit file or init rc script. The launch is failing because you are instead passing the nginx conf file, which launchctl does not recognize.

**Reply by DanielCoder834** [reply]
Thank you for the help
